src/main/java/seedu/address/logic/commands/FindCommand.java

```diff
@@ -1,43 +1,95 @@
 
-import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
 
+import java.util.List;
+
+import javafx.collections.ObservableList;
 import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 import seedu.address.model.OfficeConnectModel;
-import seedu.address.model.person.NameContainsKeywordsPredicate;
+import seedu.address.model.RepositoryModelManager;
+import seedu.address.model.mapping.AssignTask;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.NameContainsExactKeywordsPredicate;
+import seedu.address.model.person.Person;
+import seedu.address.model.shared.Id;
+import seedu.address.model.task.Task;
+
 
 /**
- * Finds and lists all persons in address book whose name contains any of the argument keywords.
- * Keyword matching is case insensitive.
+ * Finds all persons whose name contains any of the argument keywords.
+ * Keyword matching is case-insensitive.
  */
 public class FindCommand extends Command {
-
     public static final String COMMAND_WORD = "find";
+    public static final String MESSAGE_USAGE = "Format: " + COMMAND_WORD + " <PERSON_NAME>\n"
+            + "Example: " + COMMAND_WORD + " John Cena";
+
+    public static final String MESSAGE_TASK_ASSIGNED = "%1$s has been assigned to the following tasks:";
+    public static final String MESSAGE_NO_TASK_ASSIGNED = "%1$s does not have any assigned tasks.";
 
-    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all persons whose names contain any of "
-            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
-            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
-            + "Example: " + COMMAND_WORD + " alice bob charlie";
+    private final NameContainsExactKeywordsPredicate predicate;
 
-    private final NameContainsKeywordsPredicate predicate;
+    /**
+     * Creates FindCommand object with given personIndex
+     */
+    public FindCommand(NameContainsExactKeywordsPredicate predicate) {
+        requireAllNonNull(predicate);
 
-    public FindCommand(NameContainsKeywordsPredicate predicate) {
         this.predicate = predicate;
     }
 
+    /**
+     * Executes FindCommand
+     */
     @Override
-    public CommandResult execute(Model model, OfficeConnectModel officeConnectModel) {
-        requireNonNull(model);
-        model.updateFilteredPersonList(predicate);
-        return new CommandResult(
-                String.format(Messages.MESSAGE_PERSONS_LISTED_OVERVIEW, model.getFilteredPersonList().size()));
+    public CommandResult execute(Model model, OfficeConnectModel officeConnectModel) throws CommandException {
+        requireAllNonNull(model, officeConnectModel);
+
+        List<Person> personList = model.getAddressBook()
+                .getPersonList()
+                .filtered(predicate);
```

Comment:
would using the FilteredList<Person> in model give the same result as using the uniquePerson list in address book? or is it different

Reply:
Oh yes, it would. 

Using the FilteredList would only search what appears on the GUI, which means the subsequent find command cannot access the rest of the original list. Unlike mark/unmark/assign ... where we want users to only interact with the list shown. 

That's why I think that find command should search through the whole list instead of the filtered list.

Reply:
ahh I see ok thanks! I think can merge alr but need settle merge conflict